src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -0,0 +1,300 @@
+      UnsolvedClass e = iterator.next();
+      if (e.className.equals(missedClass.className)) {
+        for (UnsolvedMethod method : missedClass.methods) {
+          e.addMethod(method);
+        }
+        return;
+      }
+    }
+    missingClass.add(missedClass);
+  }
+
+  /**
+   * Based on the exception thrown by JavaParser, this method figure out which class file is missing
+   * in the source codes. This method is temporary and will be replaced by a proper SymbolSolver in
+   * the future.
+   *
+   * @param exceptionMessage the exception to be analyzed
+   * @return an instance of UnsolvedClass correlating to the exception messgae
+   */
+  public UnsolvedClass getTheMissingClass(UnsolvedSymbolException exceptionMessage) {
+    // a null cause means that the exception could not tell us which class file is missing
+    if (exceptionMessage.getCause() == null) {
+      return new UnsolvedClass("NULL", "NULL");
+    }
+    Throwable cause = exceptionMessage.getCause();
+    // This is a bit hard-coding since the Throwable instance of UnsolvedSymbolException has
+    // everything in the form of a message
+    if (cause != null && cause.getMessage() != null) {
+      String className = cause.getMessage().replace("Unsolved symbol : ", "");
+      return new UnsolvedClass(
+          className, this.classAndImportMap.getOrDefault(className, this.chosenPackage));
+    }
+    return new UnsolvedClass("NULL", "NULL");
+  }
+
+  /**
+   * The method to update synthetic files. After each run, we might have new synthetic files to be
+   * created, or new methods to be added to existing synthetic classes. This method will delete all
+   * the synthetic files from the previous run and re-create those files with the input from the
+   * current run.
+   */
+  public void updateSyntheticSourceCode() {
+    for (UnsolvedClass missedClass : missingClass) {
+      this.deleteOldSyntheticClass(missedClass);
+      this.createMissingClass(missedClass);
+    }
+  }
+
+  /**
+   * This method is to delete a synthetic class. If that synthetic class is not created yet, the
+   * method will do nothing.
+   *
+   * @param missedClass a synthetic class to be deleted
+   */
+  public void deleteOldSyntheticClass(UnsolvedClass missedClass) {
+    String classPackage = classAndImportMap.getOrDefault(missedClass.className, this.chosenPackage);
+    String filePathStr = this.rootDirectory + classPackage + "/" + missedClass.className + ".java";
+    Path filePath = Path.of(filePathStr);
+    try {
+      Files.delete(filePath);
+    } catch (IOException e) {
+      // It means that the class that has not been created in the previous run of this visitor
+    }
+  }
+
+  /**
+   * This method create a synthetic file for a class that is not in the source codes. The class will
+   * be created in the root directory of the input. All these synthetic files will be deleted when
+   * Specimin finishes its run. (TO DO: What if Specimin doesn't have the write permission?)
```

Comment:
In that case, Specimin should stop with a helpful error message (probably by throwing an informative exception).